docs/emacs.org

```diff
@@ -0,0 +1,106 @@
+  - then in Emacs, switch to js-mode, have all the bells and whistles: syntax-highlighting, autocomplete, etc.;
+  - write some javascript;
+  - finish editing, press =C-c C-c=
+    and it would paste the code back into the Dev Tools console.
+
+** Setup and customization
+
+   After Spacehammer invokes ~emacsclient~ (note that Emacs *has to be running in daemon mode*, [[https://www.gnu.org/software/emacs/manual/html_node/emacs/Emacs-Server.html][see documentation]]), it invokes emacs-lisp function(s). Those functions are in =~./hammerspoon/spacehammer.el= Emacs package. That package needs to be pre-loaded into Emacs.
+
+*** Package installation
+
+    The package currently is not available on MELPA or other public repositories, so you'd have to use ~use-package~ or other ways to load it to Emacs.
+
+*** Spacemacs
+
+    Spacemacs users can either:
+    - add the package recipe to ~dotspacemacs-additional-packages~;
+    - or create a custom Spacemacs layer;
+    - or add required pieces to an existing one.
+
+      Creating a layer is easy, you need ~packages.el~ file in a directory for your custom layer (to learn more, check [[https://github.com/syl20bnr/spacemacs/blob/develop/doc/LAYERS.org][Spacemacs documentation]])
+
+**** Spacemacs layer minimal example
+    Let's say you call the layer ~my-layer~, then the directory structure would be the following:
+
+    #+begin_src
+    ├── my-layer
+    │   └── packages.el
+    #+end_src
+
+    You place ~my-layer~ in ~dotspacemacs-configuration-layer-path~ directory of your Spacemacs config.
+
+    Here's a minimal example of ~packages.el~ that includes spacehammer.el:
+
+    - First, you need to add spacehammer to the list of packages included in the layer
+      #+begin_src emacs-lisp
+        (defconst my-layer-packages
+          '((spacehammer
+             :location (recipe                        ; Basically this telling Emacs
+                        :fetcher file                 ; where to look for the package file (spacehammer.el)
+                        :path "~/.hammerspoon/"))))
+
+
+        ;; Sometimes (depending on Emacs version and other things) that approach may not
+        ;; work. Emacs will complain about not being able to load the package. In that
+        ;; case, you can symlink the file and the directory structure for the layer has
+        ;; to be like this:
+
+        ;; .
+        ;; ├── local
+        ;; │   └── spacehammer
+        ;; │       └── spacehammer.el -> ~/.hammerspoon/spacehammer.el
+        ;; └── packages.el
+
+        ;; and the recipe would have to be something like this:
+
+        (defconst my-layer-packages
+          '((spacehammer :location local)))
+
+        ;; if you'd like to use the same Spacemacs config on different machines that
+        ;; aren't Macs, and you don't want it to complain about not finding the package
+        ;; (since Hammerspoon is not there):
+
+        (defconst my-layer-packages `(,(when (eq system-type 'darwin)
```

Comment:
Is the `(,` intentional?

Reply:
yes, it's a backquote marker https://www.gnu.org/software/emacs/manual/html_node/elisp/Backquote.html